src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<R>(items: R[], key: keyof R): GroupsMap<R> {
+  const groups: GroupsMap<R> = {};
+
+  items.forEach((item: R) => {
+    const condition: R[keyof R] = item[key];
```

Comment:
```suggestion
    const condition: stirng = item[key];
```

we already knows that R[keyof R] - string